src/types/router.ts

```diff
@@ -86,27 +92,27 @@ export type Router<
   /**
    * Registers a hook to be called before a route is entered.
    */
-  onBeforeRouteEnter: AddBeforeRouteHook,
+  onBeforeRouteEnter: AddBeforeRouteHook<TOptions>,
```

Comment:
Why do these need generics now?

Reply:
we don't need the generics, technically I could remove the generics from hooks/rejections and just have the `Router<TRoutes, TOptions>` just eslint ignore the fact that `TOptions` is unused.

Reply:
Seems like if we don't need them we shouldn't use them. Curious what this looks like without them